File: Desafios_C_Sharp/Solução_de_Problemas_Essenciais_com_C_Sharp/desafio_04.cs
/* Desafio 4 - Validação de Nota */

using System;

class Desafio {
    static void Main(string[] args) {
    
    float sum = 0;
    int counter = 0;
    bool end = false;
    
    while(!end)
    {
      float input = float.Parse(Console.ReadLine());
      
      if(input >= 0 && input <=10)
      {
        sum += input;
        counter++;
      }
      else
      {
        Console.WriteLine("nota invalida");
      }
      if(counter == 2)
      {
        float media = (sum/2);
        
        Console.WriteLine("media = " + media.ToString("N2"));
        
        end = true;
      }
    }
  }
}

File: Desafios_Javascript/Solução_de_Problemas_Essenciais/desafio_04.js
/* Desafio 4 - Validação de Nota */

let array = [];
let end = false;

while(!end){
  let input = parseFloat(gets());
  if(input >= 0 && input <=10){
    array.push(input);
  }
  else{
    console.log('nota invalida')
  }
  if(array.length === 2){
    let media = array.reduce((a, b) => (a+b)/2);
    console.log(`media = ${media.toFixed(2)}`);
    
    end = true;
  }
};